search_service/proxy/elasticsearch.py

```diff
@@ -98,7 +105,11 @@ def _get_search_result(self, page_index: int,
                 result = {}
                 for attr, val in es_payload.items():
                     if attr in model.get_attrs():
-                        result[attr] = val
+                        if attr in CLASS_MAPPING:
+                            result[attr] = [CLASS_MAPPING[attr](property_val) for property_val in val]  # type: ignore
```

Comment:
could you shed some lights on why we need to do this transformation? And is it possible that we don't do this hack given we have already ingest the badge into the ES document?

Reply:
if we don't do mapping, the badge information ends up being [{tag_name: None, tag_type: 'badge'}] when i tested it. 
Wasn't sure how to do automatic mapping since it's a more complex object 
Happy to change it if there's a smarter way I haven't thought of though!